processing/src/main/java/io/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -362,6 +387,10 @@ public void reset()
                 done = !foundMatched && (emptyRange || !baseIter.hasNext());
               }
 
+              private boolean beyondMaxRowIndex(Integer rowIndex) {
```

Comment:
The entry.getValue() is Integer, isn't better to use Integer to avoid boxing/unboxing?
>                   Map.Entry<IncrementalIndex.TimeAndDims, Integer> entry = baseIter.next();
>                   if (beyondMaxRowIndex(entry.getValue())) {
>                     continue;
>                   }